src/main/java/net/fabricmc/loom/util/DependencyProvider.java

```diff
@@ -196,27 +196,34 @@ public String getResolvedDepString() {
 				}
 			}
 
-			File root = classifierToFile.get(""); //We've built the classifierToFile map, now to try find a name and version for our dependency
+			if (dependency.getGroup() != null && dependency.getVersion() != null) {
```

Comment:
When does a `SelfResolvingDependency` have a proper group and version? Not doubting that it can, just I've never seen it outside of custom dependency types.

Reply:
I just wrote one to have correct group and version... Anyways if you put the current one with mapping provider, it will always fail because the version will be `1.0` and the version parser will parse minecraft version to be `1` and mapping version to be `0`, which apparently is the reason why all file jar mappings are broke after fudge's pr.